src/commands.rs

```diff
@@ -473,3 +485,38 @@ fn list_archive_contents(
     list::list_files(archive_path, files, list_options);
     Ok(())
 }
+
+fn check_mime_type(
+    files: &[PathBuf],
+    formats: &mut Vec<Vec<Extension>>,
+    question_policy: QuestionPolicy,
+) -> crate::Result<ControlFlow<()>> {
+    for (path, format) in files.iter().zip(formats.iter_mut()) {
+        if format.is_empty() {
+            // File with no extension
+            // Try to detect it automatically and prompt the user about it
+            if let Some(detected_format) = try_infer(path) {
+                info!("Detected file: `{}` extension as `{}`", path.display(), detected_format);
+                if user_wants_to_continue_decompressing(path, question_policy)? {
+                    format.push(detected_format);
+                } else {
+                    return Ok(ControlFlow::Break(()));
+                }
+            }
+        } else if let Some(detected_format) = try_infer(path) {
+            // File ending with extension
+            // Try to detect the extension and warn the user if it differs from the written one
+            let outer_ext = format.iter().next().unwrap();
+            if outer_ext != &detected_format {
+                info!("The file extension: `{}` differ from the detected extension: `{}`", outer_ext, detected_format);
+                if !user_wants_to_continue_decompressing(path, question_policy)? {
+                    return Ok(ControlFlow::Break(()));
+                }
+            }
+        } else {
+            // Could not detect the extension automatically
+            // Should we tell the user about it?
```

Comment:
I guess you don't need to tell the user about it because it's guaranteed that `!format.is_empty()`, right?

Reply:
That else branch is this case:
- The file has an extentiom
- We tried to detect the extention but it failed

The most probable case is that the file have an extention that we dont support

I can just add a simple warning

Reply:
What I want to write is something like:
`warning!("Could not detect the file extension, are you sure that this is a valid archive?");`

But the only problem is I'm not sure if there are some false positive or not.

Reply:
I also fear we run into some false positives, so I think in this case we should just warn the user using the `info!()` macro, and proceed normally, if our warning is wrong, at least everything still works.

Reply:
done